### PR TITLE
Updated linter to use analyzer ^2.0.0

### PR DESCRIPTION
**Pull request by usergenic**
- Look at all the things I updated.
 - [x] CHANGELOG.md has been updated

---
This change is [Reviewable](https://reviewable.io/reviews/polymer/polymer-linter/86)

### CODE REVIEW

File: src/polymer/databind-with-unknown-property.ts [hunk 1]
```diff
@@ -49,14 +49,14 @@ class DatabindWithUnknownProperty extends HtmlRule {
       const element = elements.values().next().value;
 
       const explicitlyKnownProperties =
-          new Set(element.properties.map((p) => p.name)
-                      .concat(element.methods.map((m) => m.name))
+          new Set([...element.properties.keys()]
```
Comment (justinfagnani) [inline src/polymer/databind-with-unknown-property.ts]: You can use commas with array spread:

```ts
new Set([...element.properties.keys(), ...element.methods.keys(), ...sharedProperties])
```

File: src/polymer/databinding-calls-must-be-functions.ts [hunk 2]
```diff
@@ -104,21 +104,22 @@ class DatabindingCallsMustBeFunctions extends Rule {
         continue;
       }
       const potentialMethodNames = new Set(
-          element.properties
+          [...element.properties.values()]
```
Comment (justinfagnani) [inline src/polymer/databinding-calls-must-be-functions.ts]: I'm not convinced spread is more readable than `Array.from()`

File: src/polymer/set-unknown-attribute.ts [hunk 3]
```diff
@@ -139,10 +142,10 @@ class SetUnknownAttribute extends HtmlRule {
 }
 
 function closestOption(name: string, isAttribute: boolean, element: Element) {
-  const attributeOptions = element.attributes.map((a) => a.name)
-                               .concat(Array.from(sharedAttributes.keys()));
-  const propertyOptions = element.properties.map((a) => a.name)
-                              .concat(Array.from(sharedProperties.keys()));
+  const attributeOptions = [...element.attributes.keys()].concat(
```
Comment (justinfagnani) [inline src/polymer/set-unknown-attribute.ts]: mix of spread and Array.from

**Review by justinfagnani — APPROVED** [review]

**Comment by usergenic** [comment]
Thanks, @justinfagnani updated and now waiting for CI to merge and release.

**Pull request merged**